src/Accounts/Accounts/CommonModule/ContextAdapter.cs

```diff
@@ -195,30 +195,35 @@ public object GetParameterValue(string resourceId, string moduleName, Invocation
         internal async Task<HttpResponseMessage> AuthenticationHelper(IAzureContext context, string endpointResourceIdKey, string endpointSuffixKey, HttpRequestMessage request, CancellationToken cancelToken, Action cancelAction, SignalDelegate signal, NextDelegate next, TokenAudienceConverterDelegate tokenAudienceConverter = null)
         {
             IAccessToken accessToken = await AuthorizeRequest(context, request, cancelToken, endpointResourceIdKey, endpointSuffixKey, tokenAudienceConverter);
-            var newRequest = await request.CloneWithContentAndDispose(request.RequestUri, request.Method);
-            var response = await next(request, cancelToken, cancelAction, signal);
-
-            if (response.MatchClaimsChallengePattern())
+            using (var newRequest = await request.CloneWithContent(request.RequestUri, request.Method))
```

Comment:
```suggestion
            using (var newRequest = await request.CloneWithContent(request.RequestUri, request.Method))
            using (var response = await next(request, cancelToken, cancelAction, signal))
```

Reply:
Refer to the implementation of pipeline, the original request and response will be disposed finally.